### ISSUE TITLE
Bad diagnostic spans for unpacked assignments with attribute or subscript targets

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#4413]
We now have pretty good diagnostics for unpacked assignments where every target is a `Name` expression. For example, on this code:

```py
a: int

a, b = "foo", "bar"
```

we emit this diagnostic:

<details>
<summary>Screenshot</summary>
<img width="1558" height="442" alt="Image" src="https://github.com/user-attachments/assets/08a204bc-b501-4360-a0a6-ffe2ad630c58" />
</details>

But our diagnostics are still quite poor if any target on the left-hand side is an attribute or subscript expression. On this code:

```py
class Foo:
    a: int

Foo().a, b = "foo", "bar"

wut: list[int] = [42]

wut[0], b, = "foo", "bar"
```

we emit these diagnostics:

<details>
<summary>Screenshot</summary>
<img width="2748" height="498" alt="Image" src="https://github.com/user-attachments/assets/d6121cb6-4deb-4699-baa7-8f5a81eeea3b" />
</details>

The diagnostic spans in particular are far from ideal: they don't give you any information about exactly which variable in the unpacking causes the problem.

Ideally we'd do better here, but we left it out of scope of https://github.com/astral-sh/ruff/pull/28041 since it's considerably more complex than for `Name` expressions.
